tracer/src/tracer.rs

```diff
@@ -134,16 +140,17 @@ impl TraceBlock {
 
 impl Drop for TraceBlock {
     fn drop(&mut self) {
+        // SAFETY: start is a variable that's local to the current thread
+        let start = unsafe { TRACER.get().unwrap().start };
         let trace_event = TraceEvent {
-            timestamp: self
-                .start
-                .duration_since(unsafe { TRACER.get().unwrap().start }),
+            timestamp: self.start.duration_since(start),
             event: self.event,
-            end_timestamp: Some(
-                Instant::now().duration_since(unsafe { TRACER.get().unwrap().start }),
-            ),
+            end_timestamp: Some(Instant::now().duration_since(start)),
+            // SAFETY: thread_depth() returns a number local to the current thread
             depth: unsafe { TRACER.get().unwrap().thread_depth() },
         };
+        // SAFETY: add_event and decrease_thread_depth access storage local to
+        // the current thread only
```

Comment:
This makes it sound like `thread_local!` so perhaps:

`// SAFETY: add_event and decrease_thread_depth access current thread only specific data`